Add API for popup GUI doc - docrepr.webview(obj)

techtonik opened the pull request:
![image](https://cloud.githubusercontent.com/assets/515889/9574184/f6ccb61e-4fcc-11e5-9721-e124b266789c.png)

I reduced the window size manually. The perfect size and position needs to be calculated. For now it is default size and position, which is anchored at bottom middle and is too large to my liking.

techtonik committed:
Add API for popup GUI doc - docrepr.webview(obj)

techtonik commented:
@ccordoba12 ping?

ccordoba12 commented on docrepr/__init__.py:
```diff
@@ -16,6 +16,8 @@
 Derived from spyderlib.utils.inspector and IPython.core.oinspect
 """
 
+__version__ = '0.0.1'
```
Please change version to `0.1`. `0.0.1` is too small :-)

techtonik replied:
Done.

ccordoba12 commented on docrepr/__init__.py:
```diff
@@ -24,3 +26,8 @@
+
+# This imports IPython and PySide/PyQt4
+# [ ] find a way to do delayed import
+from .webview import webview
```
This adds PyQt4/PySide as a dependency to `docrepr`, right? If that's the case, I'd like to see a different approach because I don't want `docrepr` to depend on any of them.

techtonik replied:
Fixed.

ccordoba12 commented on README.md:
```diff
@@ -25,6 +25,16 @@ Example:
     >>> docrepr.sphinxify.rich_repr(oinfo)
     c:\users\user\appdata\local\temp\docrepr\tmpwvoj3s\rich_repr_output.html
 
+`docrepr` can also show GUI window with rendered HTML documentation if you
```
Please change this text to

> `docrepr` can also show a window with the rendered representation of a docstring as an html page

techtonik replied:
Changed.

ccordoba12 commented on docrepr/webview.py:
```diff
@@ -0,0 +1,55 @@
+    try:
+        from PyQt4 import  QtGui, QtWebKit
+    except ImportError:
+        sys.stderr.write("WebView requires PySide or PyQt4 for GUI window")
```
Will this message (and the one below) show if users only do

``` python
import docrepr
```

?

techtonik replied:
Not anymore.

ccordoba12 commented:
Sorry, I totally forgot about this one. I already did a review of your work :-)

ccordoba12 commented:
Thanks for the contribution, by the way :-)

ccordoba12 commented on docrepr/webview.py:
```diff
@@ -0,0 +1,55 @@
+
+Requires PyQt4 or PySide. Original code by
+anatoly techtonik <techtonik@gmail.com> is placed in
+public domain.
```
Please remove your name and different license from here. My reasons:
1. `docrepr` authors are also all Spyder authors (because this code is derived from it) and you're already one of them.
2. You have to accept the license of the project to contribute to it.

techtonik replied:
It is not a difference license, but a note that original code submitted by me is placed into public domain and doesn't bear additional restrictions of Spyder license in case somebody will need to reuse it in their own project.

ccordoba12 replied:
But this is not a script, nor something that can be used independently of the project. It's just an additional module. So please remove your name and the public domain notice out of it. The `AUTHORS` file already mention those who have worked in the project.

This has been the work of several people, who helped to fix bugs and improve things in the code inherited from Spyder. That's why in `AUTHORS` appears the whole Spyder Development Team as author.

techtonik replied:
This is code that could be copy/pasted into different project that need a webview for their HTML stuff without reference to Spyder, and me is one of those people person. You may remove the note yourself, but I submit it as a public domain contribution. And you can remove the note, because it is public domain, but I don't want to do this myself for the reason stated - if I need the code later, I will copy/paste the code from my commit that is clearly marked as PD.

It is a common practice to take public domain code and modify it.
https://github.com/python/cpython/blob/1fe0fd9feb6a4472a9a1b186502eb9c0b2366326/Modules/_ctypes/libffi/install-sh
https://github.com/python/cpython/blob/c1e0265342dbfc710ecdd4be6fab707f7bd74dce/Lib/doctest.py

So I don't get what are you objecting of. It is not about me trying to get more credit than other people - it is just about making clear about origin of the public domain code, so people can get formal proof for it.

ccordoba12 replied:
But what's the code that _exactly_ can be copy/pasted? I'd say the only important code is the one in `show_window`. The rest can't be really copy/pasted.

So then, please add a docstring and your name to that function only.

techtonik replied:
You copy/paste the whole working example and then strip all the meat and dependencies. But as I said - it is submitted into public domain, so after you merge it, you can strip all these "copyright" notes without any legal problems. I am okay with that.

techtonik committed:
Review comments in PR#8:

- made Qt an optional dependency for docrepr
- README wording
- version field changed from 0.0.1 to 0.1

https://github.com/spyder-ide/docrepr/pull/8/

techtonik commented:
@ccordoba12 please take a look.

techtonik committed:
README.md: webview example more suitable for copy/paste

techtonik committed:
README.md: Link to actual license text

techtonik committed:
Add example screenshot

ccordoba12 commented on README.md:
```diff
@@ -25,7 +25,19 @@ Example:
+`oname` is a just a name for the object, because Python loses it when
+variable is passed to function.
+
+![webview example](example.png)
```
Please use as an example screenshot something meaningful, such as the docrepr output of `numpy.sin`

techtonik replied:
Unfortunately, `numpy` is not available for my platform - https://github.com/numpy/numpy/issues/5479 There are some unofficial binaries though, so I need some time to test them.

ccordoba12 replied:
Ok, no problem. Then please just use `range` as an object.

techtonik replied:
Done with `numpy.sin`

ccordoba12 commented on docrepr/webview.py:
```diff
@@ -0,0 +1,63 @@
+    oinfo = Inspector().info(obj, oname)
+
+    #from pprint import pprint
+    #pprint(oinfo)
```
Please remove these comments

techtonik replied:
Replaced with a reference to https://github.com/ipython/ipython/issues/8774

ccordoba12 commented on docrepr/webview.py:
```diff
@@ -0,0 +1,63 @@
+
+    path = rich_repr(oinfo)
+    #print path
+    #print sphinxify(oinfo['docstring'])
```
Please remove these comments

techtonik replied:
Done.

techtonik committed:
Replaced example from `set` to `numpy.sin` per review comments

techtonik committed:
Remove comments per review request.

ccordoba12 commented:
@techtonik, I think I'm going to decline your contribution, not because of our copyright and ownership discussions but because I don't want to add a dependency on Qt for this package (even if optional).

At the end, the purpose of docrepr is to generate web pages for docstrings :-)

techtonik commented:
> At the end, the purpose of docrepr is to generate web pages for docstrings

Or view them?

ccordoba12 commented:
But that's for projects that use `docrepr` as a library to decide how they want to show the generated output, not for the library itself.

techtonik commented:
Okay. Now that docrepr is on PyPI, it doesn't matter. BTW, the generator is extremely slow - is it possible to generate contents into memory?
